mirgecom/fluid.py

```diff
@@ -46,35 +46,172 @@
-        of :class:`~meshmode.dof_array.DOFArray` for species mass per unit volume.
-        The species mass vector has components, $\rho~Y_\alpha$, where $Y_\alpha$
-        is the vector of species mass fractions.
+        of :class:`~meshmode.dof_array.DOFArray`, or an object array with shape
+        ``(nspecies, ndim)`` respectively for scalar or vector quantities
+        correpsonding to the numspecies species mass conservation equations.
+
+    :example::
+
+        Use `ConservedVars` to access the fluid conserved variables (CV).
+
+        The vector of fluid CV is commonly denoted as $\mathbf{Q}$, and for a
+        fluid mixture with `nspecies` species and in `ndim` spatial dimenions takes
+        the form:
+
+        .. math::
+
+            \mathbf{Q} &=
+            \begin{bmatrix}\rho\\\rho{E}\\\rho{v}_{i}\\\rho{Y}_{\alpha}\end{bmatrix},
+
+        with the `ndim`-vector components of fluid velocity ($v_i$), and the
+        `nspecies`-vector of species mass fractions ($Y_\alpha$). In total, the
+        fluid system of equations has a number of equations $N_\mbox{eq}$ =
+        (`ndim` + 2 + `nspecies`).
+
+        Internally to `MIRGE-Com`, $\mathbf{Q}$ is stored as an object array
+        (:class:`numpy.ndarray`) of :class:`~meshmode.dof_array.DOFArray`, one for
+        each component of the fluid $\mathbf{Q}$, i.e. a flat object array of
+        $N\mbox{eq}$ :class:`~meshmode.dof_array.DOFArray`.
+
+        To use this dataclass for a fluid CV-specific view on the content of
+        $\mathbf{Q}$, one can call :func:`split_conserved` to get a `ConservedVars`
+        dataclass object that resolves the fluid CV associated with each conservation
+        equation::
+
+            fluid_cv = split_conserved(ndim, Q),
+
+        after which::
+
+            fluid_mass_density = fluid_cv.mass  # a DOFArray with fluid density
+            fluid_momentum_density = fluid_cv.momentum  # ndim-vector obj array
+            fluid_species_mass_density = fluid_cv.species_mass  # nspecies-vector
+
+        Examples of using `ConservedVars` as in this example in:
+
+        - :mod:`~mirgecom.boundary`
+        - :mod:`~mirgecom.euler`
+        - :mod:`~mirgecom.initializers`
+        - :mod:`~mirgecom.simutil`
+
+    :example::
+
+        Use `join_conserved` to create an agglomerated $\mathbf{Q}$ array from the
+        fluid conserved quantities (CV).
+
+        See the first example for the definition of CV, $\mathbf{Q}$, ndim,
+        nspecies, and $N_\mbox{eq}$.
+
+        Often, a user starts with the fluid conserved quantities like mass and
+        energy densities, and it is desired to glom those quantities together into
+        a *MIRGE*-compatible $\mathbf{Q}$ data structure.
+
+        For example, a solution initialization routine may set the fluid
+        quantities::
+
+            den = rho  # rho is a DOFArray with fluid density
+            velo = v  # v is an obj array of ndim DOFArrays.
```

Comment:
```suggestion
            velo = v  # v is an obj array of ndim DOFArrays denoting the components of velocity.
```

Reply:
I also wanted more text here, but ran into the line-length limit.  I will commit this and see if the linter screams at me.

Reply:
Yeah, it complained, and I could not figure out how to make the line continue inside the code block.   See how this kind-of-broken-english description grabs you (72fda48).